x/pocketcore/keeper/proof.go

```diff
@@ -50,13 +50,21 @@ func (k Keeper) SendProofTx(ctx sdk.Ctx, n client.Client, node *pc.PocketNode, p
 				ctx.Logger().Error(fmt.Sprintf("could not delete evidence is not sealed, could cause a relay leak: %s", err.Error()))
 			}
 		}
-		if evidence.NumOfProofs != claim.TotalProofs {
+
+		if evidence.NumOfProofs < claim.TotalProofs {
+			//  The node should never submit a proof whenever the number of stored relay proofs are
```

Comment:
A comment here should mention two cases for readability:

- In the case of `evidence.NumOfProofs < claim.TotalProofs`, we don't submit a proof because we cannot generate a valid merkle proof with a smaller number of relays.
- In the case of `evidence.NumOfProofs > claim.TotalProofs`, we truncate proofs of the evidence (that happens inside `GenerateMerkleProof`) and submit a proof.

I think the problem is not OOB.  If so, the fix would be to pass `evidence.NumOfProofs` to `getPseudorandomIndex`, or add a boundary check somewhere.  The problem is that a smaller evidence is not enough to generate a matching merkle proof.  That's why we don't submit a proof.

Reply:
> The problem is that a smaller evidence is not enough to generate a matching merkle proof. That's why we don't submit a proof.

Asking for my knowledge - wouldn't you be able to generate a valid proof if the pseudo-selected random index is within bounds? 

Anyhow, updated the comment. THanks for the review!